shuttle/src/runtime/task/mod.rs

```diff
@@ -204,6 +205,16 @@ impl TaskSignature {
         }
     }
 
+    pub(crate) fn new_resource(&mut self, static_create_location: &'static Location<'static>) -> ResourceSignature {
```

Comment:
A general heads up which I forgot about on your previous PR as well: from the docs for [`caller`](https://doc.rust-lang.org/std/panic/struct.Location.html#method.caller): `Returns the source location of the caller of this function. If that function’s caller is annotated then its call location will be returned, and so on up the stack to the first call within a non-tracked function body.`, AKA: I don't think we have to pass the `Location` everywhere, we could just `#[track_caller]` all the way to the public API.